adbcug_adapter/adapter.py

```diff
@@ -168,12 +163,147 @@ def arangodb_graph_to_cugraph(
         """
         graph = self.__db.graph(name)
         v_cols = graph.vertex_collections()
-        e_cols = {col["edge_collection"] for col in graph.edge_definitions()}
+        e_cols = {e_d["edge_collection"] for e_d in graph.edge_definitions()}
 
         return self.arangodb_collections_to_cugraph(
             name, v_cols, e_cols, **query_options
         )
 
+    def cugraph_to_arangodb(
+        self,
+        name: str,
+        cug_graph: CUGGraph,
+        edge_definitions: List[Json],
+        batch_size: int = 1000,
+        keyify_nodes: bool = False,
+        keyify_edges: bool = False,
+    ) -> ADBGraph:
+        """Create an ArangoDB graph from a cuGraph graph, and a set of edge
+        definitions.
+
+        :param name: The ArangoDB graph name.
+        :type name: str
+        :param cug_graph: The existing cuGraph graph.
+        :type cug_graph: cugraph.classes.graph.Graph
+        :param edge_definitions: List of edge definitions, where each edge definition
+            entry is a dictionary with fields "edge_collection",
+            "from_vertex_collections" and "to_vertex_collections"
+            (see below for example).
+        :type edge_definitions: List[adbcug_adapter.typings.Json]
+        :param batch_size: The maximum number of documents to insert at once
+        :type batch_size: int
+        :param keyify_nodes: If set to True, will create custom node keys based on the
+            behavior of ADBCUG_Controller._keyify_cugraph_node().
+            Otherwise, ArangoDB _key values for vertices will range from 0 to N-1,
+            where N is the number of cugraph nodes.
+        :type keyify_nodes: bool
+        :param keyify_edges: If set to True, will create custom edge keys based on
+            the behavior of the ADBNX_Controller._keyify_cugraph_edge().
+            Otherwise, ArangoDB _key values for edges will range from 0 to E-1,
+            where E is the number of cugraph edges.
+        :type keyify_edges: bool
+        :return: The ArangoDB Graph API wrapper.
+        :rtype: arango.graph.Graph
+
+        Here is an example entry for parameter **edge_definitions**:
+
+        .. code-block:: python
+        [
+            {
+                "edge_collection": "teach",
+                "from_vertex_collections": ["teachers"],
+                "to_vertex_collections": ["lectures"]
+            }
+        ]
+        """
+        logger.debug(f"Starting cugraph_to_arangodb('{name}', ...):")
+        for e_d in edge_definitions:
+            self.__validate_attributes(
+                "Edge Definitions", set(e_d), self.EDGE_DEFINITION_ATRIBS
+            )
+
+        self.__db.delete_graph(name, ignore_missing=True)
+        adb_graph: ADBGraph = self.__db.create_graph(name, edge_definitions)
+
+        adb_v_cols = adb_graph.vertex_collections()
+        adb_e_cols = [e_d["edge_collection"] for e_d in edge_definitions]
+
+        is_homogeneous = len(adb_v_cols + adb_e_cols) == 2
```

Comment:
I understand the purpose of `is_homogeneous`, but am not sure about the naming. Is it correct to call a graph homogeneous if it consists of exactly one vertex and one edge collection?

Reply:
Ah good point, will address that in the next commit

Reply:
> Is it correct to call a graph homogeneous if it consists of exactly one vertex and one edge collection
from what I understand this is indeed the case, it made more sense to remove `is_homogeneous` in favour of `has_one_vcol` and `has_one_ecol`